pkgs/development/libraries/glibc/NIX_LD_LIBRARY_PATH.patch

```diff
@@ -0,0 +1,47 @@
+Sometimes apps come with a shell script that rewrites LD_LIBRARY_PATH before
+launching the actual executable (common within games and apps distributed via
+Steam). To handle this, add NIX_LD_LIBRARY_PATH for use with OpenGL and FHS
+chroot environments.
+
+_dl_fatal_printf doesn't report strerror(errno) because it causes compilation
+errors that, while can be fixed, require adding weak attribute to quite a few
+symbols which results in a much larger and less maintainable patch.
+
+https://github.com/NixOS/nixpkgs/issues/21109
+
+diff -urNZ a/elf/rtld.c b/elf/rtld.c
+--- a/elf/rtld.c	2017-02-05 15:28:43.000000000 +0000
++++ b/elf/rtld.c	2017-11-04 22:40:26.438866194 +0000
+@@ -1170,9 +1170,31 @@
+   DL_SYSDEP_OSCHECK (_dl_fatal_printf);
+ #endif
+
++  char *nix_library_path = getenv ("NIX_LD_LIBRARY_PATH");
```

Comment:
I doubt that you can call `getenv()` but rather `secure_getenv` or something.

Reply:
@dezgeg Fixed, see `unsecvars.h` patch.